sdk/go/common/resource/config/object.go

```diff
@@ -0,0 +1,487 @@
+	case map[string]object:
+		for _, v := range v {
+			if v.Secure() {
+				return true
+			}
+		}
+		return false
+	case string:
+		return c.secure
+	default:
+		return false
+	}
+}
+
+// Decrypt decrypts any ciphertexts within the object and returns appropriately-shaped Plaintext values.
+func (c object) Decrypt(ctx context.Context, decrypter Decrypter) (Plaintext, error) {
+	switch v := c.value.(type) {
+	case bool:
+		return NewPlaintext(v), nil
+	case int64:
+		return NewPlaintext(v), nil
+	case float64:
+		return NewPlaintext(v), nil
+	case string:
+		if !c.secure {
+			return NewPlaintext(v), nil
+		}
+		plaintext, err := decrypter.DecryptValue(ctx, v)
+		if err != nil {
+			return Plaintext{}, err
+		}
+		return NewSecurePlaintext(plaintext), nil
+	case []object:
+		vs := make([]Plaintext, len(v))
+		for i, v := range v {
+			pv, err := v.Decrypt(ctx, decrypter)
+			if err != nil {
+				return Plaintext{}, err
+			}
+			vs[i] = pv
+		}
+		return NewPlaintext(vs), nil
+	case map[string]object:
+		vs := make(map[string]Plaintext, len(v))
+		for k, v := range v {
+			pv, err := v.Decrypt(ctx, decrypter)
+			if err != nil {
+				return Plaintext{}, err
+			}
+			vs[k] = pv
+		}
+		return NewPlaintext(vs), nil
+	default:
+		contract.Failf("unexpected value of type %T", v)
+		return Plaintext{}, nil
+	}
+}
+
+// Get gets the member value at path. The path to the receiver is prefix.
+func (c object) Get(prefix, path resource.PropertyPath) (object, bool, error) {
+	if len(path) == 0 {
+		return c, true, nil
+	}
+
+	prefix = append(prefix, path[0])
+	switch v := c.value.(type) {
+	case []object:
+		index, ok := path[0].(int)
+		if !ok || index < 0 || index >= len(v) {
+			return object{}, false, nil
+		}
+		elem := v[index]
+		return elem.Get(prefix, path[1:])
+	case map[string]object:
+		key, ok := path[0].(string)
+		if !ok {
+			return object{}, false, nil
+		}
+		elem, ok := v[key]
+		if !ok {
+			return object{}, false, nil
+		}
+		return elem.Get(prefix, path[1:])
+	default:
+		return object{}, false, nil
+	}
+}
+
+// Delete deletes the member value at path. The path to the receiver is prefix.
+func (c *object) Delete(prefix, path resource.PropertyPath) error {
+	if len(path) == 0 {
+		return nil
+	}
+
+	prefix = append(prefix, path[0])
+	switch v := c.value.(type) {
+	case []object:
+		index, ok := path[0].(int)
+		if !ok || index < 0 || index >= len(v) {
+			return nil
+		}
+		if len(path) == 1 {
+			c.value = append(v[:index], v[index+1:]...)
+			return nil
+		}
+		elem := &v[index]
+		return elem.Delete(prefix, path[1:])
+	case map[string]object:
+		key, ok := path[0].(string)
+		if !ok {
+			return nil
+		}
+		if len(path) == 1 {
+			if len(v) == 2 {
+				keys := make([]string, 0, 2)
+				for k := range v {
+					if k != key {
+						keys = append(keys, k)
+					}
+				}
+				if len(keys) == 1 && keys[0] == "secure" {
+					if _, ok := v["secure"].value.(string); ok {
+						return errSecureReprReserved
+					}
+				}
+			}
```

Comment:
This bit is confusing. Would you mind clarifying in comments what's going on here?